tests/test_dicom.py

```diff
@@ -118,28 +122,98 @@ def test_read_rgb(self, dicom_object):
+        ],
+    )
+    def test_nvjpeg_autoselect(self, request, mocker, dicom_fixture, use_nvjpeg, available, exp):
+        dicom_file = request.getfixturevalue(dicom_fixture)
+
+        # patch methods
+        a = mocker.patch("dicom_utils.dicom.nvjpeg2k_is_available", return_value=available)
+
+        # patch to use CPU instead of GPU
+        def new(dcm, *args, **kwargs):
+            if not a():
+                raise ImportError("nvJPEG not available")
+            return dcm.pixel_array
+
+        m = mocker.patch("dicom_utils.dicom.nvjpeg_decompress", side_effect=new)
+
+        # read and check
+        with pydicom.dcmread(dicom_file) as dcm:
+            read_dicom_image(dcm, use_nvjpeg=use_nvjpeg, nvjpeg_batch_size=1)
+        assert m.called == exp
+
     @pytest.mark.ci_skip  # CircleCI will not have a GPU
-    @pytest.importorskip("pynvjpeg", reason="pynvjpeg is not installed")
-    def test_nvjpeg(self, dicom_file_j2k: str, mocker):
-        BATCH_SIZE: Final[int] = 1
-        mocked_get_batch_size = mocker.patch("dicom_utils.dicom._nvjpeg_get_batch_size")
-
-        def read_image(use_nvjpeg: bool):
-            ds = pydicom.dcmread(dicom_file_j2k)
-            image = read_dicom_image(ds, use_nvjpeg=use_nvjpeg, nvjpeg_batch_size=BATCH_SIZE)
-            if use_nvjpeg and image_is_uint16(ds):
-                mocked_get_batch_size.assert_called_with(BATCH_SIZE, ds.get("NumberOfFrames", 1))
-            else:
-                mocked_get_batch_size.assert_not_called
-            return image
-
-        cpu_image = read_image(False)
-        gpu_image = read_image(True)
+    @pytest.mark.usefixtures("pynvjpeg")
+    @pytest.mark.parametrize(
+        "dicom_fixture",
+        [
+            "dicom_file_j2k_3d_uint16",
+            "dicom_file_j2k_uint16",
+            "dicom_file_jpl14",
+            "dicom_file_j2k_int16",
+        ],
+    )
+    def test_nvjpeg(self, request, dicom_fixture):
+        dicom_file_j2k = request.getfixturevalue(dicom_fixture)
+
+        ds = pydicom.dcmread(dicom_file_j2k)
+        gpu_image = read_dicom_image(ds, use_nvjpeg=True, nvjpeg_batch_size=1)
+        cpu_image = read_dicom_image(ds, use_nvjpeg=False)
 
         assert cpu_image.shape == gpu_image.shape
         assert (cpu_image == gpu_image).all()
 
 
+@pytest.mark.ci_skip  # CircleCI will not have a GPU
+@pytest.mark.usefixtures("pynvjpeg")
+@pytest.mark.parametrize(
+    "dicom_fixture",
+    [
+        "dicom_file_j2k_3d_uint16",
+        "dicom_file_j2k_uint16",
+        pytest.param("dicom_file_jpl14", marks=pytest.mark.xfail(raises=ValueError)),
+        pytest.param("dicom_file_j2k_int16", marks=pytest.mark.xfail(raises=ValueError)),
+    ],
+)
+def test_nvjpeg_decompress(request, dicom_fixture):
+    dicom_file_j2k = request.getfixturevalue(dicom_fixture)
+
+    ds = pydicom.dcmread(dicom_file_j2k)
+    gpu_image = nvjpeg_decompress(ds, batch_size=1)
+    cpu_image = ds.pixel_array
```

Comment:
We don't have to here because `nvjpeg_decompress` does not alter the input DICOM, nor does it access `pixel_array`. Thus there isn't anything pydicom could cache.